dashboard/test/ui/features/ed_programs/pd/teacher_application.feature

```diff
@@ -129,3 +129,89 @@ Scenario: Basic teacher application submission
   Then I see no difference for "Principal approval confirmation form"
   Then I close my eyes
 
+# [MEG] TODO: Remove experiment flag from the two URLs below
+Scenario: Teacher saves, re-opens, and submits an application
+  Given I create a teacher named "Severus"
+  And I am on "http://studio.code.org/pd/application/teacher?enableExperiments=teacher-application-saving-reopening"
+  And I wait until element "h1" contains text "Professional Learning Program Teacher Application"
+  And I open my eyes to test "Saving and Reopening Teacher Application"
+
+  # Saving the application
+  Then I wait until element "h3" contains text "Section 1: About You and Your School"
+  And I press the first "input[name='country']" element
+  And I press the first "input[name='completingOnBehalfOfSomeoneElse'][value='No']" element
+  And I press keys "Severus" for element "input#firstName"
+  And I press the first "button#save" element
+  Then I wait until element "p" contains text "Your progress has been saved. Return to this page at any time to continue working on your application."
```

Comment:
very nice! you've got the right pattern here -- if something is broken, we always want to catch it with a separate command like this one if we can, rather than relying on the eyes diff to catch it.

Reply:
Ah excellent––thanks for the context!